elastic/apm-server
# Fix metadata.service.* fields requirements in json schema

## Motivation/summary

Validations on metadata.service.* fields were not run correctly. This PR changes the schema to properly run validations for `v2` and `rum-v3`:
* `metadata.service.name`,`metadata.service.agent.name`: key required, value must be non-empty string , 
* `metadata.service.agent.version`: key must be sent, value must not be null
* `metadata.service.runtime.name`,`metadata.service.runtime.version`: keys must be sent, values must not be null if `metadata.service.runtime` is sent
* `metadata.service.language.name`: keys must be sent, values must not be null if `metadata.service.language` is sent


The PR also fixes an issue where `metadata.service.cloud.*` fields that are not required were not allowed to be sent as `null` values. 


## Checklist

~- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).~

I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~
- [x] get elastic APM agent devs involved to check if the fix breaks any of the current agents.

## How to test these changes
Sending payloads where `metadata.service.agent.name`,`metadata.service.agent.version`,`metadata.service.runtime.name`,`metadata.serice.runtime.version`,`metadata.service.language.name` are missing or set to `null` and ensure payload is invalid. 

## Related issues
fixes #4141
